Makefile

```diff
@@ -4,14 +4,81 @@
 DEBUG  ?= 0
 OPENMP ?= 1
 MPI    ?= 0
-CXX     = g++
+CXX     = g++ 
 MKDIR_P = mkdir -p
 RM      = /bin/rm
 
 MAKE_CPPFLAGS = -D_GNU_SOURCE -Iinclude -I${GTEST_ROOT}/include
 MAKE_CXXFLAGS = -Wall -std=c++11
 MAKE_LDFLAGS  = -L. -L${GTEST_ROOT} -lgtest
 
+
+
+ifeq ($(PE_ENV), CRAY)
+CXX = CC
+MAKE_CPPFLAGS = -Iinclude -I${GTEST_ROOT}/include
```

Comment:
We don't need to set the preprocessor flags for each compiler. All of them accept the `-D` and `-I` options.